eng/pipelines/eng-common-sync.yml

```diff
@@ -118,75 +110,25 @@ stages:
 
+                ${{ if eq(variables['Build.Reason'], 'PullRequest') }}:
                   - pwsh: |
-                      $PRData = "Azure;${{ repo }};$(Submitted.PullRequest.Number)"
-                      Add-Content -Path "$(Build.ArtifactStagingDirectory)/${{ parameters.PRDataFileName }}" -Value $PRData
-                    displayName: Write Sync PR Data to Artifact File
-                    condition: succeeded()
-                
-                - task: PublishPipelineArtifact@1
-                  condition: succeeded()
-                  displayName: Publish ${{ parameters.PRDataFileName }} 
-                  inputs:
-                    artifactName: ${{ parameters.ArtifactName }}
-                    path: $(Build.ArtifactStagingDirectory)/${{ parameters.PRDataFileName }} 
-
-  - stage: VerifyAndMerge
-    jobs:
-      - deployment: VerifyandMergeSyncPrs
-        displayName: Verify and Merge Sync PRs
-        environment: githubmerges
-
-        pool:
-          vmImage: windows-2019
-
-        strategy:
-          runOnce:
-            deploy:
-              steps:
-                - checkout: self
-
-                - download: current
-                  artifact: ${{parameters.ArtifactName}}
-                  displayName: Download ${{ parameters.PRDataFileName }}
-
-                - task: PowerShell@2
-                  displayName: 'Verify then Merge Pull Requests'
-                  inputs:
-                    targetType: filePath
-                    filePath: $(Build.SourcesDirectory)/eng/scripts/Verify-And-Merge-PRs.ps1
-                    arguments: >
-                      -PRDataArtifactPath "$(Pipeline.Workspace)/${{parameters.ArtifactName}}/${{ parameters.PRDataFileName }}"
-                      -AuthToken "$(azuresdk-github-pat)"
-                      -devOpsLogging
-                    pwsh: true
-
-                - ${{ if ne(variables['Build.Reason'], 'PullRequest') }}:
-                  - ${{ each repo in parameters.Repos }}:
-                    - pwsh: |
-                        git clone https://github.com/azure/${{ repo }} --depth 1
-                        pushd $(System.DefaultWorkingDirectory)/${{ repo }}
-                        git push origin --delete "sync-${{ parameters.DirectoryToSync }}-$(System.PullRequest.SourceBranch)-$(System.PullRequest.PullRequestNumber)"
-                        if ($lastExitCode -ne 0) {
-                          Write-Host "Failed to delete [sync-${{ parameters.DirectoryToSync }}-$(System.PullRequest.SourceBranch)-$(System.PullRequest.PullRequestNumber)] branch in ${{ repo }}"
-                          exit 1
-                        }
-                      displayName: Write Sync PR Data to Artifact File
-                      workingDirectory: $(System.DefaultWorkingDirectory)
-                      condition: succeeded()
+                      . "$(Build.SourcesDirectory)\eng\common\scripts\Invoke-GitHub-API.ps1" -AuthToken "$(azuresdk-github-pat)"
```

Comment:
Why aren't creating a wrapper script for this? It will definitely make diagnosing issues easier.